sql/src/main/java/io/crate/operation/collect/sources/InformationSchemaIterables.java

```diff
@@ -99,10 +118,30 @@ public InformationSchemaIterables(final Schemas schemas,
         return sqlFeatures;
     }
 
+    public Iterable<IngestionRuleInfo> ingestionRules() {
+        return ingestionRules;
+    }
+
     public Iterable<Void> keyColumnUsageInfos() { return keyColumnUsages; }
 
     public Iterable<Void> referentialConstraintsInfos() { return referentialConstraints; }
 
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (event.metaDataChanged() == false) {
+            return;
+        }
+        createMetaDataBasedIterables(event.state().getMetaData());
+    }
+
+    private void createMetaDataBasedIterables(MetaData metaData) {
+        RoutineInfos routineInfos = new RoutineInfos(fulltextAnalyzerResolver,
+            metaData.custom(UserDefinedFunctionsMetaData.TYPE));
+        routines = FluentIterable.from(routineInfos).filter(Objects::nonNull);
+
+        IngestionRuleInfos ingestionRuleInfos = new IngestionRuleInfos(metaData.custom(IngestRulesMetaData.TYPE));
+        ingestionRules = FluentIterable.from(ingestionRuleInfos);
```

Comment:
ingestionRuleInfos is already an iterable - shouldn't be necessary to wrap it in a FluentIterable